src/main/scala/org/broadinstitute/hail/variant/VariantSampleMatrix.scala

```diff
@@ -284,6 +284,25 @@ class VariantSampleMatrix[T](val metadata: VariantMetadata,
   def sampleVariants(fraction: Double): VariantSampleMatrix[T] =
     copy(rdd = rdd.sample(withReplacement = false, fraction, 1))
 
+  def head(n: Long): VariantSampleMatrix[T] = {
+
+    val mergeResult = (index: Int, taskResult: Int) => taskResult
+    var nVariantsPerPartition = 0
+     rdd.sparkContext.runJob(rdd,
+      (tc: TaskContext, it: Iterator[(Variant, Annotation, Iterable[T])]) => it.size,
+      Seq[Int](1),
+      (index: Int, size: Int) => nVariantsPerPartition = size
+    )
+
+    val nPartitions = math.ceil(n.toDouble/nVariantsPerPartition).toInt
+
+    copy(rdd
+        .mapPartitionsWithIndex((idx, iter) =>
+          if (idx < nPartitions) if(idx == nPartitions-1) iter.take((n % nVariantsPerPartition).toInt) else iter else Iterator())
```

Comment:
I agree with Tim: this version computes the first RDD twice (once to get the size) and then does something for every partition.  I think `sc.parallelize(rdd.take(n))` is going to be faster if the input is large and `n` is small.  The only downside is that it transmits data to/from the master, although that shouldn't be bad when `n` is small.

Best would be to implement a TakeRDD class that has a single partition and implements the logic of RDD.take to compute that single partition on a worker.  Then there is no unnecessary transmission.  I'll leave it up to Laurent which of the two he would prefer to do.

I'm fine if this is in `filtervariants head`.

I think we should have a similar command for samples.